beetsplug/fetchart.py

```diff
@@ -317,6 +317,17 @@ def get(self, album, plugin, paths):
         """Return the Cover Art Archive and Cover Art Archive release group URLs
         using album MusicBrainz release ID and release group ID.
         """
+
+        # Cover Art Archive API offers pre-resized thumbnails at several sizes.
+        # If the maxwidth config matches one of the already available sizes
+        # fetch it directly intead of fetching the full sized image and
+        # resizing it.
+        valid_thumbnail_sizes = [250, 500, 1200]
+        if plugin.maxwidth in valid_thumbnail_sizes:
+            size_suffix = "-" + str(plugin.maxwidth)
+            self.URL += size_suffix
+            self.GROUP_URL += size_suffix
```

Comment:
Aha; one last thing I noticed: it will behave somewhat strangely to modify those class fields. Instead, let's change this locally:
```suggestion
        release_url = self.URL
        group_url = self.GROUP_URL
        if plugin.maxwidth in valid_thumbnail_sizes:
            size_suffix = "-" + str(plugin.maxwidth)
            release_url += size_suffix
            group_url += size_suffix
```

…and also use those local variables in the two calls below.

Reply:
Done.